util/proxy.go

```diff
@@ -0,0 +1,27 @@
+package util
+
+import (
+	"context"
+	"net"
+	"sync"
+
+	"github.com/go-sql-driver/mysql"
+	"golang.org/x/net/proxy"
+)
+
+var (
+	setMySQLProxyOnce sync.Once
+)
+
+// SetMySQLProxyFromEnvironment sets the mysql dial specified by the proxy-related variables in the environment,
+// and makes underlying connections directly.
+// ALL_PROXY/all_proxy, NO_PROXY/no_proxy
+func SetMySQLProxyFromEnvironment() {
+	dialer := proxy.FromEnvironment()
+
+	setMySQLProxyOnce.Do(func() {
+		mysql.RegisterDialContext("tcp", func(ctx context.Context, addr string) (net.Conn, error) {
+			return dialer.Dial("tcp", addr)
```

Comment:
```
                        d, err := proxy.FromURL(...)
                        if err != nil {
                                return nil, err
                        }
                        if xd, ok := d.(proxy.ContextDialer); ok {
                                return xd.DialContext(ctx, "tcp", addr)
                        }
                        return d.Dial("tcp", addr)
```

This is what I am using. The proxy implementation actually has `DialContext`, but you need a cast.

Check [this code](https://github.com/golang/net/blob/5f55cee0dc0d/proxy/dial.go#L17-L31).

Reply:
I like it, but in this case(proxy.FromURL), I prefer to use KUBE_PROXY as the proxy URL envir var name than ALL_PROXY since ALL_PROXY may affect some other kinds of connections nobody knows.

Reply:
That is trivial. But some tools does not support `KUBE_PROXY`, because tipocket can not inject the proxy logic into their libraries. I am fine with it though, all accesses are toward the k8s cluster.

My point ist that you can cast and use `DialContext` to make use of `context.Context` from mysql package.

Reply:
> My point is that you can cast and use `DialContext` to make use of `context.Context` from mysql package.

Thanks, that's resolved on 3fe3b4c

Reply:
And another way is by adding a flag, I'm thinking about this way.

Reply:
It makes sense to me. Maybe you can keep both methods.